tss-esapi/tests/integration_tests/attributes_tests/command_code_attributes_tests.rs

```diff
@@ -0,0 +1,220 @@
+
+    assert_eq!(
+        expected_vendor_specific,
+        command_code_attributes.is_vendor_specific(),
+        "CommandCodeAttributes converted from TPMA_CC did not contain the expected value for Vendor Specific(V)"
+    );
+
+    assert_eq!(
+        expected_tpma_cc,
+        command_code_attributes.into(),
+        "CommandCodeAttributes did not convert into the expected TPMA_CC value"
+    );
+}
+
+#[test]
+fn test_conversions_vendor_specific() {
+    let expected_command_index: u16 = 0xFFFFu16; // Vendor specfic so anything goes
+    let expected_nv = true;
+    let expected_extensive = true;
+    let expected_flushed = true;
+    let expected_c_handles: u8 = 6;
+    let expected_r_handle = true;
+    let expected_vendor_specific = true;
+
+    let expected_tpma_cc = create_tpma_cc(
+        expected_command_index,
+        expected_nv,
+        expected_extensive,
+        expected_flushed,
+        expected_c_handles,
+        expected_r_handle,
+        expected_vendor_specific,
+    );
+
+    let command_code_attributes: CommandCodeAttributes = expected_tpma_cc
+        .try_into()
+        .expect("Failed to convert TPMA_CC to CommandCodeAttributes");
+
+    assert_eq!(
+        expected_command_index,
+        command_code_attributes.command_index(),
+        "CommandCodeAttributes converted from TPMA_CC did not contain the expected value for command index"
+    );
+
+    assert_eq!(
+        expected_nv,
+        command_code_attributes.nv(),
+        "CommandCodeAttributes converted from TPMA_CC did not contain the expected value for NV"
+    );
+
+    assert_eq!(
+        expected_extensive,
+        command_code_attributes.extensive(),
+        "CommandCodeAttributes converted from TPMA_CC did not contain the expected value for Extensive"
+    );
+
+    assert_eq!(
+        expected_flushed,
+        command_code_attributes.flushed(),
+        "CommandCodeAttributes converted from TPMA_CC did not contain the expected value for Flushed"
+    );
+
+    assert_eq!(
+        expected_c_handles,
+        command_code_attributes.c_handles(),
+        "CommandCodeAttributes converted from TPMA_CC did not contain the expected value for CHandles"
+    );
+
+    assert_eq!(
+        expected_r_handle,
+        command_code_attributes.r_handle(),
+        "CommandCodeAttributes converted from TPMA_CC did not contain the expected value for RHandle"
+    );
+
+    assert_eq!(
+        expected_vendor_specific,
+        command_code_attributes.is_vendor_specific(),
+        "CommandCodeAttributes converted from TPMA_CC did not contain the expected value for Vendor Specific(V)"
+    );
+
+    assert_eq!(
+        expected_tpma_cc,
+        command_code_attributes.into(),
+        "CommandCodeAttributes did not convert into the expected TPMA_CC value"
+    );
+}
+
+#[test]
+fn test_invalid_conversions_non_vendor_specific_invalid_command_index() {
+    let invalid_tpma_cc = create_tpma_cc(
+        0xFFFFu16, // Invalid command code index,
```

Comment:
I don't get it, is there something invalid about the index? Seems to be the same one as the previous test.

Reply:
I has `false` as a last argument here indicating that this should *not* be a vendor specific (compare last arguments). I agree the long list of true+falses is kind of confusing :)

Maybe moving parameters to explicit variables like in the last test would help?

Reply:
Nah, I think what confused me is that the comment seems to refer to the index, even though it's not just the index that's invalid, but the index+another parameter.

Reply:
Ack. A potential for some improvement here :)

Reply:
Well the command index ```0xFFFFu16``` is invalid if it is not vendor specific. because if it is vendor specific it can be any u16 number.

Reply:
True, but that's what I meant, that the index is not inherently invalid, that depends on whether the CC is vendor-specific (so it should be reflected in that comment)

Reply:
Yeah, that is a good point. It was reflected in the function name but that got so long that it got hard to read. I will fix it and make it clear.

Reply:
Fixed